src/main/java/newbank/server/NewBankClientHandler.java

```diff
@@ -6,46 +6,128 @@
 import java.io.PrintWriter;
 import java.net.Socket;
 
+import newbank.server.exceptions.DuplicateCustomerException;
+import newbank.server.exceptions.PasswordInvalidException;
+import newbank.server.exceptions.RequestNotAllowedException;
+
+/** The NewBankClientHandler handles all clients requests. */
 public class NewBankClientHandler extends Thread {
 
   private NewBank bank;
   private BufferedReader in;
   private PrintWriter out;
+  private CustomerID customer;
 
   public NewBankClientHandler(Socket s) throws IOException {
     bank = NewBank.getBank();
     in = new BufferedReader(new InputStreamReader(s.getInputStream()));
     out = new PrintWriter(s.getOutputStream(), true);
   }
 
+  private void handleLogin(String[] tokens) {
+    String username = "";
+    String password = "";
+
+    if (tokens.length >= 2) {
+      username = tokens[1];
+    }
+    if (tokens.length >= 3) {
+      password = tokens[2];
+    }
+
+    customer = bank.checkLogInDetails(username, password);
+
+    if (customer != null) {
+      out.println("SUCCESS: Log In Successful");
+    } else {
+      out.println("FAIL: Log In Failed");
+    }
+  }
+
+  private void handleRegisterCustomer(String[] tokens) {
+    String username = "";
+    String password = "";
+
+    if (tokens.length >= 2) {
+      username = tokens[1];
+    }
+    if (tokens.length >= 3) {
+      password = tokens[2];
+    }
+
+    try {
+      bank.addCustomer(username, password);
+
+      out.println("SUCCESS: Account created successfully.");
```

Comment:
Account in this context is a little ambiguous. `Customer account` perhaps?

Reply:
Done.